functions/src/index.ts

```diff
@@ -261,5 +262,45 @@ export const getStatistics = functions.https.onCall(async (data, context) => {
   throw new functions.https.HttpsError("invalid-argument", "invalid provider");
 });
 
+export const getUserEmotes = functions.https.onCall(async (data, context) => {
+  if (!context.auth) {
+    throw new functions.https.HttpsError("permission-denied", "missing auth");
+  }
+  const provider = data?.provider;
+  const channelId = data?.channelId;
+  if (!provider || !channelId) {
+    throw new functions.https.HttpsError(
+      "invalid-argument",
+      "missing provider, channelId"
+    );
+  }
+
+  switch (provider) {
+    case "twitch":
+      const channel = await getTwitchLogin(context.auth.uid, channelId);
+      if (!channel) {
+        return;
+      }
+      const client = await getTwitchClient(context.auth.uid, channelId);
+      await client.connect();
+
+      var retrieveEmotes = new Promise(function (resolve) {
+        client.on("emotesets", (set: string, emotes: EmoteObj) => resolve(emotes));
+      })
+
+      try {
+        const emoteInfo: any = await retrieveEmotes;
```

Comment:
The promise can be inlined:

```
try {
  const emoteInfo = await new Promise<EmoteObj>((resolve) => client.on("emotesets", (set, emotes) => resolve(emotes)));
  return { emotes: [...Object.values(emoteList)] };
} finally {
  await client.disconnect();
}
```

Reply:
This would return something like
```json
{
  "emotes": [
    [{...}, {...}]
  ]
}
```
I'll keep investigating on how to make it prettier.

Reply:
Ohh are you trying to flatten it?

```
return { emotes: Object.values(emoteList).flat() };
```

Reply:
Had to add es2019 to lib to flatten any[], not sure if thats ok or if it might cause something unexpected.

Reply:
That's fine :)